diff --git a/lib/arm/mmu.c b/lib/arm/mmu.c
index 540a1e8..5834817 100644
--- a/lib/arm/mmu.c
+++ b/lib/arm/mmu.c
@@ -87,6 +87,11 @@ static pteval_t *get_pte(pgd_t *pgtable, uintptr_t vaddr)
 	return &pte_val(*pte);
 }
 
+
+pteval_t *tr_pte(void *vaddr) {
+  return get_pte(mmu_idmap,(uintptr_t)vaddr);
+}
+
 static pteval_t *install_pte(pgd_t *pgtable, uintptr_t vaddr, pteval_t pte)
 {
 	pteval_t *p_pte = get_pte(pgtable, vaddr);
diff --git a/lib/vmalloc.h b/lib/vmalloc.h
index 8b158f5..5f4e5d5 100644
--- a/lib/vmalloc.h
+++ b/lib/vmalloc.h
@@ -3,6 +3,9 @@
 
 #include <asm/page.h>
 
+/* Pointer to page table entry from virtual address */
+extern pteval_t *tr_pte(void *vaddr) ;
+
 /* Allocate consecutive virtual pages (without backing) */
 extern void *alloc_vpages(ulong nr);
 /* Allocate consecutive and aligned virtual pages (without backing) */
